### PR TITLE
lambda: add support for container image as alternative to runtime

### LINKED ISSUE #1417: Support image_uri
### Summary

Please support lambdas from custom containers in AWS ECR.

There have been a pull request here: https://github.com/ansible-collections/community.aws/pull/772

I have created a patch, but no tests yet - thus no pull request. But in case somebody else has time to finish it, I wanted to share my work.

https://github.com/ansible-collections/amazon.aws/compare/main...manuelschneider:amazon.aws:main

Thank you very much for the modules!

### Issue Type

Feature Idea

### Component Name

lambda

### Additional Information

```yaml (paste below)
-   name: "Create and/or configure the lambda {{ lambda.key }}"
    amazon.aws.lambda:
        profile: "{{ awsProfile }}"
        region: "{{ aws_region }}"
        state: present
        name: "{{ lambda.key }}"
        role: arn:aws:iam::123456:role/myrole"
        image_uri: 123456.dkr.ecr.eu-central-1.amazonaws.com/mycontainer:stage
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by nethershaw**
##### SUMMARY
Fixes #1417
Supersedes https://github.com/ansible-collections/amazon.aws/pull/1438

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lambda

##### ADDITIONAL INFORMATION
The original PR appears to have been abandoned by its author. I have taken the liberty of salvaging it and resolving current merge conflicts as conservatively as possible.

I hope I have correctly merged the `required_if` and `required_together` arrays; it seems to make sense to me, and it seems to run correctly from my local library.

I have also targeted release 6.5.2 in the documentation snippet, if that is not too presumptive.

Please request changes. I will endeavor to oblige. Thank you.

### CODE REVIEW

File: plugins/modules/lambda.py [hunk 1]
```diff
@@ -142,6 +142,13 @@
     type: list
     elements: dict
     version_added: 5.5.0
+  image_uri:
+    description:
+      - The Amazon ECR URI of the image to use.
+      - Required (alternative to runtime zip_file and s3_bucket) when creating a function.
+      - Required when I(state=present).
+    type: str
+    version_added: 6.5.2
```
Comment (tremble) [inline plugins/modules/lambda.py:151]:
```suggestion
    version_added: 7.1.0
```

6.x is now bug/security fixes only, we can add this to 7.1 without a problem.

**Review by tremble — CHANGES_REQUESTED** [review]
While adding integration tests for this would be good, I don't think it's practical at this point since we'd need an image we could safely pull and use...

**Review by tremble — APPROVED** [review]

File: plugins/modules/lambda.py [hunk 2]
```diff
@@ -533,12 +540,21 @@ def _s3_args(s3_bucket, s3_key, s3_object_version):
     return code
 
 
+def _image_args(image_uri):
+    if not image_uri:
+        return {}
+
+    code = {'ImageUri': image_uri }
```
Comment (tremble) [inline plugins/modules/lambda.py:547]:
```suggestion
    code = {'ImageUri': image_uri}
```

File: plugins/modules/lambda.py [hunk 3]
```diff
@@ -600,20 +618,30 @@ def main():
...
+        ['zip_file', 's3_object_version'],
+        ['image_uri', 'zip_file'],
+        ['image_uri', 'runtime'],
+        ['image_uri', 'handler'],
+        ['image_uri', 's3_key'],
+        ['image_uri', 's3_bucket'],
+        ['image_uri', 's3_object_version'],
     ]
 
+    required_by = { "runtime": "handler" }
```
Comment (tremble) [inline plugins/modules/lambda.py:632]:
```suggestion
    required_by = {"runtime": ["handler"]}
```

**Review by alinabuzachis — APPROVED** [review]
@nethershaw Only and small note, but LGTM otherwise! Thank you!

**Pull request merged**